tests/visualization_tests/test_timeline.py

```diff
@@ -0,0 +1,62 @@
+from io import BytesIO
+from typing import List
+
+import pytest
+
+import optuna
+from optuna.study.study import Study
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+from optuna.visualization._plotly_imports import go
+import optuna.visualization._timeline
+from optuna.visualization._timeline import _get_timeline_info
+
+
+@pytest.mark.parametrize(
+    "n",
+    [0, 3],
+)
+def test_get_timeline_info(n: int) -> None:
+    study = optuna.create_study()
+    for _ in range(n):
+        study.add_trial(
+            optuna.trial.create_trial(
+                params={"x": 0.5},
+                distributions={"x": optuna.distributions.FloatDistribution(0, 1)},
+                value=0,
+                state=optuna.trial.TrialState.COMPLETE,
+            )
+        )
+    assert len(_get_timeline_info(study).bars) == n
```

Comment:
Might be good to assert the `_TimelineInfo` itself.

Reply:
Please see [other visualization tests](https://github.com/optuna/optuna/blob/master/tests/visualization_tests/test_contour.py#L207) for example.

Reply:
I agree with the principle that we should follow the existing style of tests. In this particular case, however, I don't think it is not a good way to pre-write down and test the exact same intermediate data. The reason is that `study.add_trial` gets and puts in the time information on the fly.